src/Provider/AbstractProvider.php

```diff
@@ -285,8 +286,32 @@ public function getAuthorizationUrl(array $options = [])
             'response_type'   => $options['response_type'],
             'approval_prompt' => $options['approval_prompt'],
         ];
+    }
 
-        return $this->urlAuthorize().'?'.$this->httpBuildQuery($params, '', '&');
+    /**
+     * Builds the authorization URL's query string.
+     *
+     * @param array $params Query parameters
+     * @return string Query string
+     */
+    protected function getAuthorizationQuery(array $params)
+    {
+        return http_build_query($params);
+    }
+
+    /**
+     * Returns the authorization URL for given options.
+     *
+     * @param array $options
+     * @return string
+     */
+    public function getAuthorizationUrl(array $options = [])
+    {
+        $base   = $this->urlAuthorize();
+        $params = $this->getAuthorizationParameters($options);
+        $query  = $this->getAuthorizationQuery($params);
+
+        return $base.'?'.$query;
```

Comment:
Good point. I thought about it but figured it's not likely that there is a case where no query string is present in the auth url. How else would it work? Even then, an empty query string (or trailing ?) wouldn't affect the URL.